deploy/sophosxg.sh

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env sh
+#
+# This deploy script deploys to a Sophos XG appliance
+# DEPLOY_SOPHOSXG_HOST="<NO DEFAULT - REQUIRED - host:port>"
+# DEPLOY_SOPHOSXG_USER="<NO DEFAULT - REQUIRED - string>"
+# DEPLOY_SOPHOSXG_PASSWORD="<NO DEFAULT - REQUIRED - string>"
+# DEPLOY_SOPHOSXG_NAME="domain"
+# DEPLOY_SOPHOSXG_PFX_PASSWORD="s0ph0sXG"
+# DEPLOY_SOPHOSXG_HTTPS_INSECURE="1"
+
+########  Public functions #####################
+
+#action pfx user password name pfxpass host
+sophosxg_do_req() {
+
+  # does curl request to upload certificate to sophos appliance
+
+  # check number of args
+  [ $# -eq 7 ] || return 1
+
+  # set vars
+  _do_req_action="$1"
+  _do_req_pfx="$2"
+  _do_req_user="$3"
+  _do_req_password="$4"
+  _do_req_name="$5"
+  _do_req_pfxpass="$6"
+  _do_req_host="$7"
+
+  # create temp file for xml
+  _info "Creating request XML"
+  _do_req_xml="$(_mktemp)"
+  if [ ! -f "$_do_req_xml" ]; then
+    _err "Error creating temp file for XML"
+    return 1
+  fi
+
+  # create xml request
+  echo "
+<Request>
+  <Login>
+    <Username>${_do_req_user}</Username>
+    <Password>${_do_req_password}</Password>
+  </Login>
+  <Set operation=\"${_do_req_action}\">
+    <Certificate>
+      <Action>UploadCertificate</Action>
+      <Name>${_do_req_name}</Name>
+      <Password>${_do_req_pfxpass}</Password>
+      <CertificateFormat>pkcs12</CertificateFormat>
+      <CertificateFile>certificate.p12</CertificateFile>
+      <PrivateKeyFile></PrivateKeyFile>
+    </Certificate>
+  </Set>
+</Request>
+" >"$_do_req_xml"
+
+  # dont verify certificate if HTTPS_INSECURE was set
+  if [ "$Le_Deploy_sophosxg_https_insecure" = "1" ] || [ "$HTTPS_INSECURE" ]; then
+    _sophosxg_curl="$_sophosxg_curl --insecure"
+  fi
+
+  # do request with curl
+  $_sophosxg_curl --silent -F "reqxml=<$_do_req_xml" -F "file=@$_do_req_pfx;filename=certificate.p12" "https://$_do_req_host/webconsole/APIController?" | grep -q '<Status code="200">'
+  ret=$?
+
+  # remove xml file
+  rm -f "$_do_req_xml"
+
+  return $ret
+}
+
+#domain keyfile certfile cafile fullchain
+sophosxg_deploy() {
+  _cdomain="$1"
+  _ckey="$2"
+  _ccert="$3"
+  _cca="$4"
+  _cfullchain="$5"
+
+  # check for curl first
+  if _exists "curl"; then
```

Comment:
why curl is required ?
Can you please use `_get()` or `_post()` functions instead ?

Reply:
I've updated my branch to remove this requirement.

I am now building the required "multipart/form-data" request body so it can be done via "_post"